### PR TITLE
New consent manager view states: DoNotSellDisclosure, AcceptOrRejectAll

### PR DESCRIPTION
**Pull request by michaelfarrell76**
- closes https://transcend.height.app/T-17147

## Playground has new do not sell button

Screen Shot 2022-08-31 at 9 42 52 PM

## New view state = DoNotSellDisclosure & GPC

Screen Shot 2022-08-31 at 9 43 36 PM

## New view state = DoNotSellDisclosure & No GPC
Screen Shot 2022-08-31 at 9 46 53 PM

## New view state = AcceptOrRejectAll

Screen Shot 2022-09-01 at 8 24 28 AM

## Updated view state = AcceptAll

Screen Shot 2022-09-01 at 8 24 23 AM

## Video

https://user-images.githubusercontent.com/10264973/187833739-78057dec-2e76-42c9-8646-61151f560b95.mov

### CODE REVIEW

File: src/components/DoNotSellAcknowledgement.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  /** Function to change viewState */
+  handleSetViewState: HandleSetViewState;
+}): JSX.Element {
+  const { airgap } = useAirgap();
+  const { formatMessage } = useIntl();
+
+  const handleOptOut: JSX.MouseEventHandler<HTMLButtonElement> | undefined = (
+    event: JSX.TargetedEvent<HTMLButtonElement, MouseEvent>,
+  ): void => {
+    event.preventDefault();
```
Comment (michaelfarrell76) [inline src/components/DoNotSellAcknowledgement.tsx:32]: @eligrey the new design for this modal requires us to set the consent preferences on footer click. im not sure we have access to an event here, how can we update the consent here without an event? or do we need to pass the event through the transcend.showConsentManager function?

Comment (eligrey) [reply src/components/DoNotSellAcknowledgement.tsx:32]: If by 'footer' you mean 'within the modal', then we definitely should be able to capture the click here.

File: src/components/AcceptOrRejectAll.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    | undefined = (
+    event: JSX.TargetedEvent<HTMLButtonElement, MouseEvent>,
+  ): void => {
+    event.preventDefault();
+    const purposeTypes = airgap.getPurposeTypes();
+    const consent: TrackingConsent = {};
+    Object.keys(purposeTypes).forEach((purpose) => {
+      consent[purpose] = true;
+    });
+    airgap.setConsent(event, consent);
```
Comment (eligrey) [inline src/components/AcceptOrRejectAll.tsx:42]:
```suggestion
    airgap.optIn(event);
```

File: src/components/AcceptOrRejectAll.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    | undefined = (
+    event: JSX.TargetedEvent<HTMLButtonElement, MouseEvent>,
+  ): void => {
+    event.preventDefault();
+    const purposeTypes = airgap.getPurposeTypes();
+    const consent: TrackingConsent = {};
+    Object.keys(purposeTypes).forEach((purpose) => {
+      consent[purpose] = false;
+    });
+    airgap.setConsent(event, consent);
```
Comment (eligrey) [inline src/components/AcceptOrRejectAll.tsx:53]:
```suggestion
    airgap.optOut(event);
```

File: src/components/AcceptAll.tsx [hunk 4]
```diff
@@ -31,17 +28,20 @@ export default function AcceptAll({
...
+    | undefined = (
+    event: JSX.TargetedEvent<HTMLButtonElement, MouseEvent>,
+  ): void => {
+    event.preventDefault();
+    const purposeTypes = airgap.getPurposeTypes();
+    const consent: TrackingConsent = {};
+    Object.keys(purposeTypes).forEach((purpose) => {
+      consent[purpose] = true;
+    });
+    airgap.setConsent(event, consent);
```
Comment (eligrey) [inline src/components/AcceptAll.tsx:42]:
```suggestion
    airgap.optIn(event);
```

File: src/components/DoNotSellAcknowledgement.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    // Confirm current consent
+    airgap.setConsent(event, {
+      ...airgap.getConsent().purposes,
+      [Purpose.SaleOfInfo]: false,
+    });
+    console.log(airgap.getConsent());
```
Comment (eligrey) [inline src/components/DoNotSellAcknowledgement.tsx:38]: Any call to `setConsent()` will confirm consent (persist existing values & set `.confirmed = true`). Also, you can modify individual keys in `setConsent()`.

```suggestion
    // Confirm current consent
    airgap.setConsent(event, {
      [Purpose.SaleOfInfo]: false,
    });
```

File: src/components/DoNotSellAcknowledgement.tsx (hunk 1 above)
Comment (michaelfarrell76) [reply src/components/DoNotSellAcknowledgement.tsx:32]: its not within the modal. its the footer on say, the athetic website. and it calls transcend.showConsentManager() on click

File: src/components/AcceptAll.tsx [hunk 6]
```diff
@@ -37,12 +31,7 @@ export default function AcceptAll({
     event: JSX.TargetedEvent<HTMLButtonElement, MouseEvent>,
   ): void => {
     event.preventDefault();
-    const purposeTypes = airgap.getPurposeTypes();
-    const consent: TrackingConsent = {};
-    Object.keys(purposeTypes).forEach((purpose) => {
-      consent[purpose] = true;
-    });
-    airgap.setConsent(event, consent);
+    airgap.optIn(event);
```
Comment (michaelfarrell76) [inline src/components/AcceptAll.tsx:34]: @eligrey did you make this change?

Comment (eligrey) [reply src/components/AcceptAll.tsx:34]: yeah. it's functionality equivalent 👍

**Review by eligrey — DISMISSED** [review]

**Review by bencmbrook — CHANGES_REQUESTED** [review]
making small styling tweaks before merge

File: src/components/AcceptOrRejectAll.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    event: JSX.TargetedEvent<HTMLButtonElement, MouseEvent>,
+  ): void => {
+    event.preventDefault();
+    airgap.optOut(event);
+    handleSetViewState('close');
+  };
+
+  return (
+    <ColumnContent>
+      <Title align="left">{formatMessage(messages.consentTitle)}</Title>
```
Comment (bencmbrook) [inline src/components/AcceptOrRejectAll.tsx:51]: OK to merge but this message isn't ready for prime time.

**Review by bencmbrook — DISMISSED** [review]

File: src/components/AcceptOrRejectAll.tsx (hunk 7 above)
Comment (michaelfarrell76) [reply src/components/AcceptOrRejectAll.tsx:51]: do you have a preferred message? 

we are also adding message overrides for customers to change it as they desire

Comment (bencmbrook) [reply src/components/AcceptOrRejectAll.tsx]: (routing to legal)

**Review by bencmbrook — APPROVED** [review]

**Review by eligrey — APPROVED** [review]

**Pull request merged**